tcp-ballerina/client.bal

```
@@ -44,20 +44,58 @@ public client class Client {
         return externWriteBytes(self, data);
     }
  
-    // remote function writeBlocksFromStream(stream<byte[]> dataStream) returns Error? { }
+    # Sends the given stream of data to the connected remote host.
+    # ```ballerina
+    # tcp:Error? result = socketClient->writeBytes("msg".toBytes());
+    # ```
+    #
+    # + data - The data need to be sent to the connected remote host
+    # + return - () or else a `tcp:Error` if the given data can't be sent
+    remote function writeBytesFromStream(stream<byte[]> dataStream) returns Error? {
+        var iterator = dataStream.iterator();
+        Error? writeError = ();
+        while(true) {
+            record {|byte[] value;|}? bytes = iterator.next();
+            if (bytes is record {|byte[] value;|}) {
+                check self->writeBytes(bytes.value);
+                if (writeError is Error) {
+                    break;
+                }
+            } else {
+                _ = dataStream.close();
```

Shouldn't we need to return this error?

---

yes, updated with return